doc/api/next_api_changes/2019-04-02-AL.rst

```diff
@@ -0,0 +1,8 @@
+Deprecations
+````````````
+
+Setting ``Axis.major.locator``, ``Axis.minor.locator``, ``Axis.major.formatter``
+or ``Axis.minor.formatter`` to an object that is not a subclass of `Locator` or
+`Formatter` (respectively) is deprecated.  Note that these attributes should
+usually be set using `Axis.set_major_locator`, `Axis.set_minor_locator`, etc.
```

Comment:
Should they be set at all directly on the ticker? I don't think so at least for locators, because `Axis.set_major_locator` ensures that the formatter knows the locator, which is not guaranteed when setting `Axis.major.locator`.

Reply:
Well, you get to argue about https://github.com/matplotlib/matplotlib/pull/13826/files#diff-52127080e2783464f529df684d82d476R1319 with @tacaswell then :)

Reply:
I don't see the point. The linked diff does not assign to `axis.major.locator` or `axis.major.formatter`?

Reply:
It does `getattr(self.minor.locator, "remove_overlaps", True)` instead of the simpler `self.minor.locator.remove_overlaps` (note that `remove_overlaps` *is* in the base Locator class) specifically to handle the case of a locator not inheriting from Locator.

Reply:
Ok, I see that we jump through some hoops to support non-Locator instances as locator.

But `_set_locator()` would be living in `TickHelper`, which hopefully nobody has replaced. AFAICS non-Locator objects would still work.

Reply:
Sorry, I don't get what you're saying about TickHelper.  Did you mean Ticker (which is not the same)?